db/db.go

```diff
@@ -26,6 +26,14 @@ const (
 	REMOTE_DATASET = "remote"
 )
 
+var (
+	transactions       = map[int]ReadTransaction{}
```

Comment:
Make these owned by `DB`.

Reply:
Made the connection own them since the mapping is only needed for the rpc